src/screens/DiscoveryScreenV2/elements/DiscoveryTitleSeparator.js

```diff
@@ -1,35 +1,78 @@
 import * as React from 'react';
-import {Text} from 'react-native';
-
+import {Animated, Text, View} from 'react-native';
+import ArrowDown from '../../../../assets/icons/arrow-down.svg';
 import {fonts} from '../../../utils/fonts';
 import {COLORS} from '../../../utils/theme';
 
 /**
  * @typedef {Object} DiscoveryTitleSeparatorProp
  * @property {String} text
+ * @property {Boolean} showArrow
+ * @property {Boolean} [rotateArrow]
  */
 /**
  *
- * @param {DiscoveryTitleSeparatorProp} prop
+ * @param {DiscoveryTitleSeparatorProp} props
  */
-const DiscoveryTitleSeparator = (prop) => {
-  const {text} = prop;
+const DiscoveryTitleSeparator = (props) => {
+  const {text, showArrow, rotateArrow = false} = props;
+
+  // Animation value for rotation
+  const rotation = React.useRef(new Animated.Value(0)).current;
+
+  // Interpolating the animation value to create a rotation style
+  const rotate = rotation.interpolate({
+    inputRange: [0, 1],
+    outputRange: ['0deg', '180deg'] // Rotate 180 degrees
+  });
+
+  // Trigger rotation animation if `rotateArrow` is true
+  React.useEffect(() => {
+    if (rotateArrow) {
+      Animated.timing(rotation, {
+        toValue: 1,
+        duration: 300, // Adjust as needed
+        useNativeDriver: true
+      }).start();
+    } else {
+      // Reset to default rotation
+      Animated.timing(rotation, {
+        toValue: 0,
+        duration: 300,
+        useNativeDriver: true
+      }).start();
+    }
+  }, [rotateArrow]);
+
   return (
-    <Text
+    <View
       style={{
-        fontSize: 12,
-        lineHeight: 18,
-        fontFamily: fonts.poppins[600],
-        color: COLORS.white,
-        paddingLeft: 20,
-        paddingRight: 20,
-        paddingTop: 11,
-        paddingBottom: 11,
-        backgroundColor: COLORS.gray110
-        // marginTop: 15,
+        backgroundColor: COLORS.gray110,
+        justifyContent: 'space-between',
+        alignContent: 'center',
+        alignItems: 'center',
+        flexDirection: 'row',
+        paddingRight: 20
       }}>
-      {text}
-    </Text>
+      <Text
+        style={{
+          fontSize: 12,
+          lineHeight: 18,
+          fontFamily: fonts.poppins[600],
+          color: COLORS.white,
+          paddingLeft: 20,
+          paddingRight: 20,
+          paddingTop: 11,
+          paddingBottom: 11
+        }}>
+        {text}
+      </Text>
+      {showArrow && (
+        <Animated.View style={{transform: [{rotate}]}}>
+          <ArrowDown />
+        </Animated.View>
+      )}
+    </View>
```

Comment:
<img src="https://avatars.githubusercontent.com/in/347564?s=41" alt="Image description" width="20" height="20">   CodeRabbit

The component rendering looks good. The conditional rendering of the `ArrowDown` component based on the `showArrow` prop is correctly implemented. The styles applied to the components seem appropriate. However, consider extracting inline styles into a StyleSheet object for better performance and readability.

```diff
   return (
    <View style={styles.container}>
      <Text style={styles.text}>
        {text}
      </Text>
      {showArrow && (
        <Animated.View style={{transform: [{rotate}]}}>
          <ArrowDown />
        </Animated.View>
      )}
    </View>
  );
};
78:
const styles = StyleSheet.create({
  container: {
    backgroundColor: COLORS.gray110,
    justifyContent: 'space-between',
    alignContent: 'center',
    alignItems: 'center',
    flexDirection: 'row',
    paddingRight: 20
  },
  text: {
    fontSize: 12,
    lineHeight: 18,
    fontFamily: fonts.poppins[600],
    color: COLORS.white,
    paddingLeft: 20,
    paddingRight: 20,
    paddingTop: 11,
    paddingBottom: 11
  }
});
```


<!-- This is an auto-generated comment by OSS CodeRabbit -->